realm/realm-library/src/main/java/io/realm/internal/Context.java

```diff
@@ -17,135 +17,22 @@
 package io.realm.internal;
 
 import java.lang.ref.ReferenceQueue;
-import java.util.ArrayList;
-import java.util.List;
 
+// Currently we free native objects in two threads, the SharedGroup is freed in the caller thread, others are freed in
```

Comment:
Related test cases have been added to core. https://github.com/realm/realm-core/blob/master/test/test_destructor_thread_safety.cpp